### PR TITLE
Enable XLA on TensorFlow ROCm port

### PR DESCRIPTION
**Pull request by whchung**
PR to enable XLA on TensorFlow ROCm port

### CODE REVIEW

**Comment by whchung** [comment]
@parallelo I've revised the PR which contains logic to make XLA for ROCm built properly. Since XLA is disabled by default when we configure TensorFlow build, I'm wondering how to change that in `ci_build`?

**Comment by parallelo** [comment]
Good question.  Not totally certain yet, but it seems like this line would be a likely candidate:  https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/blob/develop-upstream/tensorflow/tools/ci_build/xla/linux/rocm/run_py3.sh#L32

**Comment by whchung** [comment]
Looks like as long as "--config=xla" is set at build time ( https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/blob/develop-upstream/tensorflow/tools/ci_build/xla/linux/rocm/run_py3.sh#L41 ) we should be fine?

**Comment by whchung** [comment]
@parallelo I'd like reproduce the build error reported by CI locally. Could you let me now if `tensorflow/tools/ci_build/xla/linux/run_py3.sh` the script I should execute?

**Comment by parallelo** [comment]
Currently, CI is running this command from the base of `tensorflow-upstream`:  
```
tensorflow/tools/ci_build/ci_build.sh ROCM ./tensorflow/tools/ci_build/linux/rocm/run_py3_core.sh
```
The `ci_build.sh` command builds a ROCm docker image, starts the container and runs the provided script.  The `run_py3_core.sh` script kicks of a tensorflow build and run unit tests.  

We have not tried with the `tensorflow/tools/ci_build/xla/linux/run_py3.sh` script yet.  Perhaps its use of `--config=xla` would help, but not certain yet.

**Comment by parallelo** [comment]
In `xla/linux/run_py3.sh`, we might also need an updated `configure` command to enable XLA.

**Comment by whchung** [comment]
CI log shows:

```
[31m[1mERROR: [0m/var/lib/jenkins/workspace/tensorflow-upstream-unit-tests/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/external/llvm/BUILD:446:1: Traceback (most recent call last):
	File "/var/lib/jenkins/workspace/tensorflow-upstream-unit-tests/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/external/llvm/BUILD", line 447
		gentbl(name = (target["lower_name"] + "..."), <4 more arguments>)
	File "/var/lib/jenkins/workspace/tensorflow-upstream-unit-tests/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/external/org_tensorflow/third_party/llvm/llvm.bzl", line 32, in gentbl
		native.genrule(name = ("%s_%s_genrule" % (name,...)), <5 more arguments>)
genrule rule 'amdgpu_target_gen__gen_tgt_intrinsic_genrule' in package '' conflicts with existing genrule rule, defined at /var/lib/jenkins/workspace/tensorflow-upstream-unit-tests/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/external/llvm/BUILD:447:5
```
I did solve this in my local repository but turns out I inadvertently changed offending `bazel` script *INSIDE* `bazel` cache instead of the original source repository. I've just amended the PR.

File: configure.py [hunk 1]
```diff
@@ -1493,7 +1493,7 @@ def main():
   set_build_var(environ_cp, 'TF_NEED_KAFKA', 'Apache Kafka Platform',
                 'with_kafka_support', False, 'kafka')
   set_build_var(environ_cp, 'TF_ENABLE_XLA', 'XLA JIT', 'with_xla_support',
-                False, 'xla')
+                True, 'xla')
```
Comment (whchung) [inline configure.py]: @parallelo do we want to enable XLA by default, just like what we did in TF 1.3, or have it disabled by default?

**Comment by whchung** [comment]
test log shows:
```
2018-06-07 15:10:46.927359: F tensorflow/compiler/tf2xla/xla_op_registry.cc:264] Check failed: it != registry.backends_.end() Unknown backend XLA_GPU_JIT
```
let me look into it.

**Pull request merged**